R/seqCluster.R

```diff
@@ -105,33 +125,41 @@ seqCluster <- function(.data, .dist, .perc_similarity, .nt_similarity, .fixed_th
       graph_from_data_frame() %>%
       clusters() %>%
       .$membership %>%
-      melt())
+      melt() %>%
+      suppressWarnings())
     result_multi <- seq_clusters %>%
       map2(., seq_length[!singleseq_flag], ~ .x %>%
-        mutate(length_value = map_chr(.y, ~ ifelse(all(.x == .x[1]),
-          yes = .x[1],
-          no = glue("range_{min(.x)}:{max(.x)}")
-        )))) %>%
+        mutate(
+          length_value = map_chr(.y, ~ ifelse(all(.x == .x[1]),
+            yes = .x[1],
+            no = glue("range_{min(.x)}:{max(.x)}")
+          ))
+        )) %>%
       map2(., protocluster_names[!singleseq_flag], ~ rownames_to_column(.x, var = "Sequence") %>%
         group_by(value, length_value) %>%
         mutate(Cluster = paste0(.y, "_length_", length_value, "_cluster_", cur_group_id())) %>%
         ungroup() %>%
-        select(Sequence, Cluster)) %>%
-      map_df(~.x)
-    res <- rbind(result_single, result_multi)
-    colnames(res) <- c(matching_col, "Cluster")
-    res[grouping_cols] <- str_split(str_split(res[["Cluster"]],
-      pattern = "_", simplify = TRUE
-    )[, 1],
-    pattern = "/", simplify = TRUE
-    )[, seq_along(grouping_cols)]
+        select(Sequence, Cluster))
+    if (!all(is.na(grouping_cols))) {
+      result_multi %<>% map2_df(., pmap(group_values, data.frame)[!singleseq_flag], ~ cbind(.x, .y))
+      res <- rbind(result_single, result_multi)
+      res[grouping_cols] <- str_split(str_split(res[["Cluster"]],
+        pattern = "_", simplify = TRUE
+      )[, 1],
+      pattern = "/", simplify = TRUE
+      )[, seq_along(grouping_cols)]
+    } else {
+      result_multi %<>% map_df(., ~.x)
+      res <- rbind(result_single, result_multi)
+      colnames(res) <- c(matching_col, "Cluster")
+    }
     return(res)
   }
   clusters <- map(.dist, ~ graph_clustering(.x, threshold_fun = .threshold_fun))
   if (!all(map2_lgl(clusters, .data, ~ nrow(.x) == nrow(.y)))) {
     warning("Number of sequence provided in .data and .dist are not matching!")
   }
   # supress messages because join spams about joining by matching_col is done
-  result_data <- map2(.data, clusters, ~ left_join(.x, .y) %>% suppressMessages())
+  result_data <- map2(.data, clusters, ~ left_join(.x, .y, by = grouping_cols) %>% suppressMessages())
```

Comment:
Looks like this join / map results in separating Sequence column into Sequence.x and Sequence.y